feat: added `created` hook to directives (#719)
vuejs/docs-next: src/guide/custom-directive.md

A directive definition object can provide several hook functions (all optional):

> - `created`: called before bound element's parent component props are applied. This is useful in cases where the directive needs to attach event listeners before parent component's prop or template listeners.

Reviewer: ```suggestion
- `created`: called before the bound element's attributes or event listeners are applied. This is useful in cases where the directive needs to attach event listeners that must be called before normal `v-on` event listeners.
```
